boa3_test/tests/examples_tests/test_ico.py

```diff
@@ -1,328 +1,515 @@
-                                         transferred_amount, None,
-                                         signer_accounts=[self.OTHER_ACCOUNT_1])
-        self.assertEqual(True, result)
-        self.assertEqual(0, len(engine.get_events('transfer')))
-        balance_after = self.run_smart_contract(engine, path, 'balanceOf', self.OTHER_ACCOUNT_1)
-        self.assertEqual(balance_after, balance_before)
-
-        result = self.run_smart_contract(engine, path, 'transferFrom',
-                                         self.OWNER_SCRIPT_HASH, self.OTHER_ACCOUNT_1, self.OTHER_ACCOUNT_2, 0, None,
-                                         signer_accounts=[self.OTHER_ACCOUNT_1])
-        self.assertEqual(True, result)
-        self.assertEqual(0, len(engine.get_events('transfer')))
-
-        balance_originator_before = self.run_smart_contract(engine, path, 'balanceOf', self.OWNER_SCRIPT_HASH)
-        balance_sender_before = self.run_smart_contract(engine, path, 'balanceOf', self.OTHER_ACCOUNT_1)
-        balance_receiver_before = self.run_smart_contract(engine, path, 'balanceOf', self.OTHER_ACCOUNT_2)
-        result = self.run_smart_contract(engine, path, 'transferFrom',
-                                         self.OWNER_SCRIPT_HASH, self.OTHER_ACCOUNT_1, self.OTHER_ACCOUNT_2,
-                                         transferred_amount, None,
-                                         signer_accounts=[self.OTHER_ACCOUNT_1])
-        self.assertEqual(False, result)
-
-        balance_originator_after = self.run_smart_contract(engine, path, 'balanceOf', self.OWNER_SCRIPT_HASH)
-        self.assertEqual(balance_originator_before, balance_originator_after)
-        balance_sender_after = self.run_smart_contract(engine, path, 'balanceOf', self.OTHER_ACCOUNT_1)
-        self.assertEqual(balance_sender_before, balance_sender_after)
-        balance_receiver_after = self.run_smart_contract(engine, path, 'balanceOf', self.OTHER_ACCOUNT_2)
-        self.assertEqual(balance_receiver_before, balance_receiver_after)
+        balance_before_1 = runner.call_contract(path, 'balanceOf', owner_script_hash)
+        invokes.append(runner.call_contract(path, 'transferFrom',
+                                            owner_script_hash, other_account_1_script_hash, owner_script_hash,
+                                            transferred_amount, None))
+        expected_results.append(True)
+        balance_after_1 = runner.call_contract(path, 'balanceOf', owner_script_hash)
+
+        balance_before_2 = runner.call_contract(path, 'balanceOf', owner_script_hash)
+        invokes.append(runner.call_contract(path, 'transferFrom',
+                                            owner_script_hash, other_account_1_script_hash, other_account_1_script_hash,
+                                            transferred_amount, None))
+        expected_results.append(True)
+        balance_after_2 = runner.call_contract(path, 'balanceOf', owner_script_hash)
```

Comment:
shouldn't this balance be from `other_account_1_script_hash` instead of `owner_script_hash`?

Reply:
not in this case because the transfer is being triggered by `transferFrom`
other_account_1 was authorized to transfer owner funds, so it's the owner's balance that's going to be changed